src/regex.c

```diff
@@ -107,65 +110,78 @@ void free_regex(void)
 		domain->regexmatch = REGEX_UNKNOWN;
 	}
 
-	// Return early if we don't use any regex
-	if(regex == NULL)
+	// Return early if we don't use any regex filters
+	if(regex[REGEX_WHITELIST] == NULL &&
+	   regex[REGEX_BLACKLIST] == NULL)
 		return;
 
-	// Disable blocking regex checking and free regex datastructure
-	for(int index = 0; index < num_regex; index++)
+	// Free regex datastructure
+	for(int regexid = 0; regexid < 2; regexid++)
 	{
-		if(regexconfigured[index])
+		for(int index = 0; index < num_regex[regexid]; index++)
 		{
-			regfree(&regex[index]);
-
-			// Also free buffered regex strings if in regex debug mode
-			if(config.debug & DEBUG_REGEX)
+			if(regexconfigured[regexid][index])
 			{
-				free(regexbuffer[index]);
-				regexbuffer[index] = NULL;
+				regfree(&regex[regexid][index]);
+
+				// Also free buffered regex strings if in regex debug mode
+				if(config.debug & DEBUG_REGEX)
+				{
+					free(regexbuffer[regexid][index]);
+					regexbuffer[regexid][index] = NULL;
+				}
 			}
 		}
-	}
 
-	// Free array with regex datastructure
-	free(regex);
-	regex = NULL;
-	free(regexconfigured);
-	regexconfigured = NULL;
+		// Free array with regex datastructure
+		if(regex[regexid] != NULL)
+		{
+			free(regex[regexid]);
+			regex[regexid] = NULL;
+		}
+		if(regexconfigured[regexid] != NULL)
+		{
+			free(regexconfigured[regexid]);
+			regexconfigured[regexid] = NULL;
+		}
 
-	// Reset counter for number of regex
-	num_regex = 0;
+		// Reset counter for number of regex
+		num_regex[regexid] = 0;
+	}
 }
 
-void read_regex_from_database(void)
+static void read_regex_table(unsigned char regexid)
 {
+	// Get database ID
+	unsigned char databaseID = (regexid == REGEX_BLACKLIST) ? REGEX_BLACK_LIST : REGEX_WHITE_LIST;
```

Comment:
The difference between these enums is a little confusing. Perhaps the DB one should have `DB` or `TABLE` added to it?